src/Permanent/MediumPermanentPlainTextFormatter.php

```diff
@@ -83,10 +83,18 @@ private function generateWeekScheme(array $openingHoursData): string
 
         $weekScheme = $this->getWeekScheme($weekDaysOpen, $this->formatter);
 
+        $isFirstPeriodASingleDay = array_pop($weekScheme);
```

Comment:
How does this work exactly? Do you not need to check the return value to see if it's a single day? And `array_pop` returns the last element, not the first one?

Reply:
`array_pop()` returns the last element and removes it from the array and the returning `getWeekScheme()` always has a boolean as the last element.
But now that I look at my code, I think it would be better to just create a second function in the trait `MediumPermanentWeekScheme`.